lib/results.js

```diff
@@ -139,8 +144,8 @@ function encodeResult(res, count) {
     output += (res.ok ? 'ok ' : 'not ok ') + count;
     output += res.name ? ' ' + res.name.toString().replace(/\s+/g, ' ') : '';
 
-    if (res.skip) output += ' # SKIP';
-    else if (res.todo) output += ' # TODO';
+    if (res.skip) output += ' # SKIP' + (typeof res.skip === 'string' ? ' ' + res.skip : '');
```

Comment:
did this not require any test changes? if not, can we add tests that illustrate this change?

Reply:
Adding these tests I found a new problem:
``` js
tape('incomplete test with explanation', function (t) {
  // var platform = process.platform; // something like this needed
  var platform = 'win32';

  t.fail('run sh', { skip: platform === 'win32' });
  t.fail('run openssl', { skip: platform === 'win32' && 'can\'t run on windows platforms' });
  t.pass('this runs');
  t.end();
});
```

Even though `skip` is passed in `extra` the tests fail with error, IMO this needs another fix (and tests).

Probably these lines here
https://github.com/substack/tape/blob/c6c4aceccf52fb189732b95fd049b4e110d7868f/lib/test.js#L223-L227

will need to be changed to
``` js
ok = ok || res.skip; // <- goes before line 223
```

Reply:
lol, i wouldn't even expect `skip` to work in an assertion, only in a `test` or `t.test` call.

Reply:
Well this line is related to any `t.assert`, so 🤷🏻‍♂️

Reply:
> expect skip to work

To be used?

Reply:
I’ve never passed options into an assertion - I’m surprised it does anything.

Reply:
It does actually work, and as per TAP individual explanations need this. Anyways I've removed this from current branch. I will perhaps try to make another PR for it and we can discuss it their.

I've added my final tests and patches. This PR is done for, please review.